config/config.go

```diff
@@ -222,3 +241,27 @@ func BuildConfig() (*Config, error) {
 	return c, nil
 	// NOTE: what happens if there no config file and also no env vars setted?
 }
+
+// getVcsRevision gets the buildInfo vcs revision if available.
+func getVcsRevision() string {
+	if x, ok := debug.ReadBuildInfo(); ok {
+		rev := ""
+		mod := ""
+		for _, s := range x.Settings {
+			switch s.Key {
+			case "vcs.revision":
+				if len(s.Value) == 40 { // is the commin hash length
```

Comment:
This check seems very weak. What are the possible values of `vcs.revision`? I'd use a regular expression here.

Reply:
Ok, done in 537475f